Fix discover_elements

lilioid opened the pull request:
## Proposed changes
This PR includes two fixes. Their description can be extracted from the respective commit messages.

## Related issues
I have already mentioned the necessity of the fix in 2c9de3ef06aad42e93ed4f10766367c22d54924d with @timonegk 

## Necessary checks
- [x] Update package version
- [x] Run `catkin build`
- [ ] Write documentation
- [ ] Create issues for future work
- [x] Test on your machine
- [ ] Test on the robot
- [x] Put the PR on our Project board

lilioid committed:
fix discovery of non stack elemnt classes

this adds an additional check to discover_elements so that only classes
are discovered which actually are subclasses of AbstractStackElement

lilioid committed:
fix discover_elements assumption about existing src folder

previously dicover_elements assumed decisions and actions to be defined
under a src/ folder and that they are importable relative to that src/
folder. This is not always the case (for example in tests) and has now
been fixed

lilioid committed:
bump version to v0.3.5

lilioid committed:
implement element discovery for directories

timonegk reviewed (changes requested):
The changes do not work when there are two modules with the same name, for example in teh body behavior, there is `actions/timer.py` and `decisions/timer.py`. Even though `discover_elements` is called once with the path to the `actions` folder and once with the path for the `decisions` folder, the `actions/timer.py` is used both times. 
To reproduce, run `roslaunch bitbots_bringup visualization.launch`.

lilioid committed:
fix module discovery from __init__ if path is directory

lilioid committed:
fix register_actions and decision subclass restrictions

previously register_actions and register_decisions would register
classes which were not actually actions and decisions. This is now fixed
by checking whether the candidate element subclasses
AbstractActionElement or AbstractDecisionElement

lilioid commented:
> The changes do not work when there are two modules with the same name, for example in teh body behavior, there is `actions/timer.py` and `decisions/timer.py`. Even though `discover_elements` is called once with the path to the `actions` folder and once with the path for the `decisions` folder, the `actions/timer.py` is used both times.
> To reproduce, run `roslaunch bitbots_bringup visualization.launch`.

I cannot reproduce this. It might be due to the two additional changes I have since introduced.
